Packs/McAfee_ESM/Integrations/McAfee_ESM_v2/McAfee_ESM_v2_test.py

```diff
@@ -246,6 +248,65 @@ def test_edit_case(mocker):
     assert len(result['eventList']) > 0
 
 
+@freeze_time('2022-10-18T16:46:25Z')
+def test_alarm_to_incidents(mocker):
+    """
+    Given: last run object.
+
+    When: running fetch incidents.
+
+    Then: return correct incidents based on alerts time.
+
+    """
+    def fetch_alarams(since: str = None, start_time: str = None, end_time: str = None, raw: bool = False):
+        if type(start_time) == str:
+            start_time = datetime.strptime(start_time, McAfeeESMClient.demisto_format)
+        all_alarms = [alarm for alarm in alarms if
+                      datetime.strptime(alarm.get('triggeredDate'), McAfeeESMClient.demisto_format)
+                      > start_time]
+        return None, None, all_alarms
+
+    params = {
+        "url": "https://example.com",
+        "insecure": True,
+        "credentials": {
+            "identifier": "TEST",
+            "password": "TEST"
+        },
+        "version": "11.3",
+        'fetchTime': '2022-10-18 10:06:26.563999',
+        'startingFetchID': 0}
+    alarms = [{
+        'id': 1,
+        'triggeredDate': '2022-10-18T10:06:26Z'
+    },
+        {
+            'id': 2,
+            'triggeredDate': '2022-10-18T11:06:26Z'
+    }
+    ]
+
+    mocker.patch('McAfee_ESM_v2.parse_date_range', return_value=['', ''])
+    mocker.patch.object(McAfeeESMClient, '_McAfeeESMClient__login', return_value={})
+    mocker.patch.object(McAfeeESMClient, '_McAfeeESMClient__request', return_value={})
+    mocker.patch.object(McAfeeESMClient, 'fetch_alarms', return_value=(None, None, []))
+    mocker.patch.object(demisto, 'setLastRun')
+    mocker.patch.object(demisto, 'incidents')
+
+    mocker.patch.object(demisto, 'getLastRun', return_value={'alarms': {'time': arg_to_datetime('3 days')}})
+
+    client = McAfeeESMClient(params)
+    client.fetch_incidents(params=params)
+    last_run = demisto.setLastRun.call_args[0][0]
```

Comment:
let's wrap all this in a separate function mock_fetch_run_without_results that returns the new last_run